src/robot/client.ts

```diff
@@ -336,6 +341,36 @@ export class RobotClient implements Robot {
          * read and then write to 'peerConn', even after we have awaited/paused.
          */
         this.peerConn = webRTCConn.peerConnection; // eslint-disable-line require-atomic-updates
+        this.peerConn.addEventListener('iceconnectionstatechange', () => {
+          /*
+           * TODO: are there any disconnection scenarios where we can reuse the
+           * same connection and restart ice?
+           *
+           * All connection loss scenarios I tested seem to result in the peer
+           * connection getting closed, so restarting ice is not a valid way to
+           * recover.
+           */
+          if (this.peerConn?.iceConnectionState === 'closed') {
+            let retries = 0;
+            console.debug('connection closed, will try to reconnect');
+            void backOff(() =>
+              this.connect().then(
```

Comment:
is connect ensuring no others can try connecting at the same time?

Reply:
it's not, let me add that guard

Reply:
actually wait, I think we're okay since `connect` itself guards and exists early if we are connecting.